ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveInBetweenExpandRule.java

```diff
@@ -152,7 +153,7 @@ private RexInBetweenExpander(RexBuilder rexBuilder) {
     public RexNode visitCall(final RexCall call) {
       switch (call.getKind()) {
       case SEARCH: {
-        return new SearchTransformer<>(rexBuilder, call).transform().accept(this);
+        return new SearchTransformer<>(rexBuilder, call, RexUnknownAs.UNKNOWN).transform().accept(this);
```

Comment:
Is this rule expand in and between predicates in Filter operator? Should `RexUnknownAs.FALSE` be used in that case?

Reply:
The RexInBetweenExpander is also used for Project and Join operators so if we want to change the unknown context we should probably generalize a bit this class. Not sure if it will make a huge difference in the plans but I can try it out if you want.

Reply:
Raised https://github.com/apache/hive/pull/5906 as follow-up. Let's see how that goes.